### PR TITLE
Expose internals used with other style guides

### LINKED ISSUE #974: Export of non-exported functions to enable other style guides to pass CRAN checks
Please consider exporting the following functions:

- [ ] is_function_call()
- [ ] previous_non_comment()
- [ ] scope_normalize()

Context: I'm working on {[styler.equals](https://github.com/Robinlovelace/styler.equals)} based on @lorenzwalthert's very handy template: https://github.com/lorenzwalthert/styler.yours

As mentioned in #860 tests do not pass due to these lines: 

[`  if (styler:::is_function_call(pd)) {`](https://github.com/lorenzwalthert/styler.yours/blob/07b6689532aab92b1c60f106f9daf0cbb1fed6e3/R/rules-line-break.R#L11)

https://github.com/lorenzwalthert/styler.yours/blob/07b6689532aab92b1c60f106f9daf0cbb1fed6e3/R/rules-line-break.R#L14

https://github.com/lorenzwalthert/styler.yours/blob/ea4eb50e79886c5bf58dec51dff76aafbcdd4249/R/core.R#L21

@MichaelChirico advocated in https://github.com/r-lib/styler/issues/340#issuecomment-1207569513 to export those functions to make it easier to for people building alternative styles building on this great package to release them to the masses.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Re-implementation of abandoned #1037, which had complex merge conflicts.

Closes #974.

# Progress tracker

- [x] Export `compute_parse_data_nested()` helper
- [x] Export `is_*()` helpers
- [x] Add examples to new exports
- [x] Check function and argument names are okay (since these are now user-facing)
- [x] Code clean-up
- [x] ~~Update NEWS~~ (`{fledge}` is supposed to take care of this)
- [x] Code review

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Hi @IndrajeetPatil just following-up on this, is there any way I can help progress? Will be great to be able to have `styler.*` packages building on this!

File: R/expr-is.R [hunk 1]
```diff
@@ -1,38 +1,68 @@
...
-#' @describeIn pd_is Checks whether `pd` contains an expression wrapped in
-#'   curly brackets.
-#' @keywords internal
+#' @describeIn pd_is Checks whether `pd` contains an expression wrapped in curly brackets.
+#' @examples
+#' code <- "if (TRUE) { 1 }"
+#' pd <- compute_parse_data_nested(code)
+#' is_curly_expr(pd)
+#' is_curly_expr(pd$child$`17`$child$`14`)
+#'
```
Comment (IndrajeetPatil) [inline R/expr-is.R:18]: The blank lines are deliberate. 

Without them, all examples are squished together like a centipede, and it's hard to tell where one ends and the other starts.

File: R/expr-is.R [hunk 2]
```diff
@@ -44,56 +74,68 @@ is_function_call <- function(pd) {
...
-#' @keywords internal
-is_function_dec <- function(pd) {
+#' @examples
+#' code <- "foo <- function() NULL"
+#' pd <- compute_parse_data_nested(code)
+#' is_function_declaration(pd)
+#' is_function_declaration(pd$child$`12`$child$`11`)
+#'
+#' @export
+is_function_declaration <- function(pd) {
```
Comment (IndrajeetPatil) [inline R/expr-is.R:84]: I think this is a better name than `is_function_dec()`.

File: R/expr-is.R [hunk 3]
```diff
@@ -1,38 +1,68 @@
...
-#' @keywords internal
-is_cond_expr <- function(pd) {
+#' @examples
+#' code <- "if (TRUE) x <- 1 else x <- 0"
+#' pd <- compute_parse_data_nested(code)
+#' is_conditional_expr(pd)
+#' is_conditional_expr(pd$child$`24`)
+#'
+#' @export
+is_conditional_expr <- function(pd) {
```
Comment (IndrajeetPatil) [inline R/expr-is.R:47]: I think this is a better name than `is_cond_expr()`.

**Comment by IndrajeetPatil** [comment]
Hi @Robinlovelace, I am done on my end. Now I await code review. Once that's done, I think this should be good to go and might even be part of the next release (cf. #1044).

**Comment by Robinlovelace** [comment]
That is a great update, thanks. :crossed_fingers: for the next release!

File: R/expr-is.R (hunk 3 above)
Comment (lorenzwalthert) [reply R/expr-is.R:47]: I think we should keep it as is (or change all the `*_expr()` functions).

Comment (IndrajeetPatil) [reply R/expr-is.R:47]: But why do we need to change all the `*_expr()` functions? Their current names are expressive as they are.

Comment (lorenzwalthert) [reply R/expr-is.R:47]: no sorry I read your comment wrongly. I thouhg you wanted `*_expression()` everywhere, but that is (obviously) not the case.

Comment (IndrajeetPatil) [reply R/expr-is.R:47]: I see.

Okay, then this should be ready for your review. I don't have any additional changes in mind.

**Comment by lorenzwalthert** [comment]
@IndrajeetPatil I saw the review request. Please do not re-request a review from me. I don't forget about them, but I don't have time to look into it now but I will so in due time.

**Review by lorenzwalthert — APPROVED** [review]

**Comment by lorenzwalthert** [comment]
@IndrajeetPatil  can you have a look at my last commit? After this one is approved by you, I will prepare the release as discussed in #1044.

**Comment by IndrajeetPatil** [comment]
> can't request a review from you for some reason

The one who made PR can't review it themselves. That's why you can't assign me to review my own PR.

> can you have a look at my last commit 

Your changes look good to me. Thanks.

**Pull request merged**